### PR TITLE
fix(experimental elaborator): Fix globals which use function calls

### PR DESCRIPTION
**Pull request by jfecher**
# Description

## Problem\*

Resolves an issue where we would give a type error if a global ever used a function call in its definition.

## Summary\*

This one was more difficult to fix than it seemed since fixing the main bug revealed progressively more bugs.

## Additional Context

https://github.com/noir-lang/noir/pull/5170 is included here as well to help me see that all tests are now passing.

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by jfecher** [comment]
I messed up something in the merge with master since I'm getting a lot of the following errors now:
```
misplaced #[no_predicates] attribute on unconstrained function basic_checks. Only allowed on constrained functions
```
Will fix tomorrow.

**Review by michaeljklein — CHANGES_REQUESTED** [review]

File: compiler/noirc_frontend/src/tests.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ pub(crate) fn get_program(src: &str) -> (ParsedModule, Context, Vec<(Compilation
             &mut context,
             program.clone().into_sorted(),
             root_file_id,
-            false,
+            true,
```
[the remark this answers is not in the document]
Comment (jfecher) [reply compiler/noirc_frontend/src/tests.rs:84]: This was part of the CI changes to test the elaborator. Unfortunately for the frontend tests I couldn't easily test the elaborator and the existing frontend without duplicating them all. I changed them to only use the elaborator here but can change it back if we'd prefer that.

[the remark this answers is not in the document]
Comment (michaeljklein) [reply compiler/noirc_frontend/src/tests.rs:84]: Yes, please keep the normal tests on for now.

[the remark this answers is not in the document]
Comment (jfecher) [reply compiler/noirc_frontend/src/tests.rs]: Reverted

**Pull request merged**